standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -903,13 +1033,13 @@ public Database getJDODatabase(String name) throws NoSuchObjectException {
    * @throws NoSuchObjectException
    */
   @Override
-  public boolean alterDatabase(String dbName, Database db)
+  public boolean alterDatabase(String catName, String dbName, Database db)
     throws MetaException, NoSuchObjectException {
 
     MDatabase mdb = null;
     boolean committed = false;
     try {
-      mdb = getMDatabase(dbName);
```

Comment:
This code seems to be broken. You have not started transaction yet. getMDatabase can throw exception, in which case you'll get to the finally block which will try to rollback non-existent transaction and throw another exception because of that. 

I think all this code should be moved out of try-block.